tests/unit/test_sagemaker_steps.py

```diff
@@ -313,11 +392,13 @@ def test_training_step_creation_with_placeholders(pca_estimator):
             },
             'RoleArn': EXECUTION_ROLE,
             'HyperParameters': {
-                'feature_dim': '50000',
-                'num_components': '10',
-                'subtract_mean': 'True',
+                'HyperParamA.$': "$$.Execution.Input['HyperParamA']",
+                'HyperParamB.$': "$$.Execution.Input['HyperParamB']",
                 'algorithm_mode': 'randomized',
-                'mini_batch_size': '200'
+                'feature_dim': 50000,
+                'mini_batch_size': 200,
+                'num_components.$': "$$.Execution.Input['num_components']",
```

Comment:
nit: almost didn't see that this was being tested. perhaps make it easier for future contributors by adding a comment either here or where we set the `num_components` - since the estimator code isn't close to the test (locality) and this file is gigantic, might be a useful clue for others who only look at tests.

Reply:
good idea! will do